### DISCUSSION TITLE
Documentation: Supabase SSR example doesn't work

### DISCUSSION THREAD

**Discussion by arjunkomath** [supabase/supabase#20220] [Questions]
Following the code from the documentation (https://supabase.com/docs/guides/auth/server-side/creating-a-client?environment=get-server-side-props), I'm facing several problems:

~~**1. LSP errors, `appendHeader` method doesn't exist.**~~ (it was because @types/node was outdated.)

<img width="2443" alt="SCR-20240107-oyyx" src="https://github.com/supabase/supabase/assets/2555067/eb43d1e1-4485-4bc5-a329-43cd9fbd851a">

I had to modify this to make the set cookie work.

**2. After logging in, none of the client side queries work, the promise just doesn't resolve, the queries are simply stuck.**

**3. Same situation with logout, `await supabase.auth.signOut()` doesn't nothing.**

Is there any working example for Supabase SSR integration with Next 13/14 pages router?

Update: 
- Found the fix for 1

**Comment by silentworks** [comment]
This step should be taken only if you are using the pages router, can you confirm you are using the pages router and not the app router?

**Reply by arjunkomath** [reply]
Yes, I'm using pages router. I tried both next 13/14 version also. If I recall correctly getServerSideProps doesn't exist in App router.

**Reply by arjunkomath** [reply]
I found the problem with `appendHeader`, it was because `@types/node` was outdated. But client side queries still don't work.

**Reply by charislam** [reply]
Hey @arjunkomath, glad you found the fix for the first one. I'd need to see a bit more example code to suggest solutions for the second, do you have examples of where you make client-side queries?

(And I'm guessing you've checked Dev Tools to make sure the cookie is set correctly and is being sent alongside the requests?)

**Reply by arjunkomath** [reply]
I can see the cookies being set, but I managed to get client site auth working by using `@supabase/auth-helpers-react` and including the `SessionContextProvider` in `_app.js` file. I have no clue if this is the right way to approach this, but it seems to work. 🤷‍♂️

**Comment by opack** [comment]
@arjunkomath Hi! I'm having the same issue with the `signOut` promise not resolving 😩 The client is signed out correctly (the auth cookie is correctly cleared and if I refresh the page then the client is not signed in), but the promise never resolves 😭 Thus, I cannot proceed and redirect my user to the sign-in page.
Did you find a solution to this problem? I'm having this with v0.4.0 btw, and not with 0.3.0.

**Reply by arjunkomath** [reply]
@opack You can find my implementation here: https://github.com/techulus/changes-page

I don’t know if it’s the most idiomatic solution, but it seems to be working.

**Reply by opack** [reply]
Unfortunately, I don't see anything special about the way you handle the `signOut` compared to the way I do it. You return a function that is called when the user logs out. I don't see why it works any better in this case. I tested it anyway, just in case, but it doesn't work :-(
Thanks anyway.

**Reply by opack** [reply]
I finally resolved my problem. After browsing the Discord for the same issue, I ended up on [this thread](https://discord.com/channels/839993398554656828/1148935771055734885). I posted my solution [there](https://discord.com/channels/839993398554656828/1148935771055734885/1275757166212681739), but for convenience for future readers, I'll copy my solution here:

I just tried to invalidate only when *not* signing out:
```typescript
onMount(() => {
        const { data } = supabase.auth.onAuthStateChange(async (event, newSession) => {
            if (event !== 'SIGNED_OUT' && newSession?.expires_at !== session?.expires_at) {
                await invalidate('supabase:auth')
            }
        })

        return () => data.subscription.unsubscribe()
    })
```
and now it works correctly! I think that the `invalidate` short-circuits (in some way :person_shrugging:) the current rendering/processing and thus does not let the current things ending correctly, triggering again the `load` functions.
I don't think there is any security issue with not calling `invalidate` on sign out because the auth cookie is removed. Anyway, if I try to access a private page after signout, I cannot, which is expected.
Hope this helps!
